Forta-Bots-Challenge-2/src/agent.ts

```diff
@@ -0,0 +1,35 @@
+import { Finding, HandleTransaction, TransactionEvent } from "forta-agent";
+import { UNISWAP_V3_FACTORY_ADDRESS, SWAP_EVENT } from "./constants";
+import FactoryFetcher from "./factory.fetcher";
+import { providers } from "ethers";
+import { createFinding } from "./findings";
+import { Interface } from "@ethersproject/abi";
+
+export const provideHandleTransaction = (
+  swapEvent: string,
+  uniswapFactoryAddress: string,
+  provider: providers.Provider
+): HandleTransaction => {
+  return async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
+    const fetcher: FactoryFetcher = new FactoryFetcher(provider);
+
+    await Promise.all(
+      // filter txEvent logs for Swap event
+      txEvent.filterLog([swapEvent]).map(async (log) => {
+        // attempt factory call on pool contract the swap occurred on
+        const factoryAddress = await fetcher.getFactory(log.address, txEvent.blockNumber);
+
+        // add finding for swaps that took place on pools created by Uniswap V3 factory
+        if (factoryAddress === uniswapFactoryAddress.toLowerCase()) {
```

Comment:
Hey @msacco8, the issue with the current implementation is that any pool could return Uniswap's factory address even if it's not a valid one.

As a challenge, I would suggest you find an alternative solution to tackle this.

Reply:
This is key here, and because of that, it could affect your bot's implemented.

Because of that, I'm not reviewing it _super_ thoroughly, as it could still change (potentially drastically)